apps/vue-app/src/app/modules/posts/components/PostForm.vue

```diff
@@ -0,0 +1,133 @@
+              <span v-show="!formData.title" class="form-text text-danger"> Error </span>
+            </div>
+            <div class="form-group pb-3">
+              <label>Description</label>
+              <textarea class="form-control" rows="2" v-model="formData.description"></textarea>
+              <span v-show="!formData.description" class="form-text text-danger"> Error </span>
+            </div>
+            <div class="form-group pb-3">
+              <label>Category</label>
+              <select class="form-select" v-model="formData.categoryId">
+                <option v-for="category in categories" :key="category.id" :value="category.id">{{ category.name }}</option>
+              </select>
+              <span v-show="!formData.categoryId" class="form-text text-danger"> Error </span>
+            </div>
+            <div class="form-group">
+              <label>URL of the image</label>
+              <input type="text" class="form-control" v-model="formData.image" />
+              <span v-show="!formData.image" class="form-text text-danger"> Error </span>
+            </div>
+          </form>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-secondary" data-bs-dismiss="modal" ref="btnCloseModal" @click="resetFormData">Cancel</button>
+          <button class="btn btn-primary" @click="createPost">Save</button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { getCategories } from '../helpers/categories.js';
+import { createPost, updatePost } from '../helpers/posts.js';
+import { ref } from 'vue';
+import { store } from '../store/store';
+import showAlert from '../helpers/alerts.js';
+
+const btnCloseModal = ref(null);
+
+export default {
+  mixins: [ showAlert ],
+  /*✅ Activity 5: Add created hook */
+  created() {
+    this.getCategories();
+  },
+  methods: {
+    async getCategories() {
+      this.categories = await getCategories();
+    },
+    async createPost() {
+      const { title, description, categoryId, image } = this.formData;
+      if (title != '' && description != '' && image) {
+        const categoryFound = this.categories.filter((category) => category.id == categoryId);
+        this.formData.category = categoryFound[0];
+
+        if(this.action === 'Edit'){
+          await updatePost(this.formData)
+          this.showAlert('success', 'Post Updated')
+        } else {
+          this.formData.id = Math.random().toString(16).slice(2);
+          await createPost(this.formData);
+          this.showAlert('success', 'Post Created')
+        }
+
+        this.resetFormData();
+        this.$refs.btnCloseModal.click();
+      }
+    },
+    resetFormData() {
+      this.formData = {
+        id: '',
+        title: '',
+        description: '',
+        category: {},
+        categoryId: '',
+        image: '',
+        comments: []
+      };
+      this.action = 'Create',
+      store.setPostEditing(null);
```

Comment:
Obtuvé un error por esta línea. Tienes que hacer referencia al store con this.store

También por buena practica, debes agregar el store dentro de la data option